### PR TITLE
Remove the need to call modifier on exported modifier

### PR DESCRIPTION
**Pull request by madnificent**
It was previously deemed necessary to call Modifier.modifier on the exported modifier class.  We can remove this requirement based on the input of @chancancode in #7.  This commit implements the behaviour as we understood it from this issue.

The deprecation message indicates version 1.0.0 will drop support for calling Modifier.modifier, this has not been discussed and is subject to change.

The examples of the README have been updated to remove the deprecated Modifier.modifier call.

The tests have been duplicated to contain both the calls with Modifier.modifier as well as those without.  The tests have been updated mechanically with Emacs Macro's rather than being smart about them to keep them obvious.

### CODE REVIEW

File: addon/-private/modifier-classic.js
Comment (sukima) [inline addon/-private/modifier-classic.js:21]: Has this been replaced with a non-underscored method (via an import)? If so, should we make that change in this PR?

File: tests/integration/modifier-managers/oo-modifiers-classic-test.js
Comment (sukima) [inline tests/integration/modifier-managers/oo-modifiers-classic-test.js:37]: Are we supposed to still use that method here?

File: tests/integration/modifier-managers/oo-modifiers-classic-test.js
Comment (sukima) [inline tests/integration/modifier-managers/oo-modifiers-classic-test.js:507]: Should we be testing that `Modifier.modifier` has been deprecated?

File: tests/integration/modifier-managers/oo-modifiers-native-test.js
Comment (sukima) [inline tests/integration/modifier-managers/oo-modifiers-native-test.js:501]: Great job with adding so many more tests!!!

**Review by sukima — APPROVED** [review]

File: addon/-private/modifier-classic.js
Comment (madnificent) [reply addon/-private/modifier-classic.js:21]: Let's separate it out so we don't block this PR.

File: tests/integration/modifier-managers/oo-modifiers-classic-test.js
Comment (madnificent) [reply tests/integration/modifier-managers/oo-modifiers-classic-test.js:37]: There's duplication in the tests now.  We call those with Modifier.modifier too, to ensure nothing breaks.

File: tests/integration/modifier-managers/oo-modifiers-classic-test.js
Comment (madnificent) [reply tests/integration/modifier-managers/oo-modifiers-classic-test.js:507]: I've asked how to do this in Ember Discord#topic-testing.  If there's a reasonable response on how to test this, I'll add that test, otherwise I'll press that merge button.

File: tests/integration/modifier-managers/oo-modifiers-classic-test.js
Comment (sukima) [inline tests/integration/modifier-managers/oo-modifiers-classic-test.js:18]: Wow, Ember gives you a register function but not an unregister function?!

Since this is essentially a global state in a test how do you feel about moving the registration to a test helper? Maybe something like this:

```js
setupDeprecartionTest(hooks);
```

Maybe look like:

```js
// tests/helpers/deprecations.js
import { registerDeprecationHandler } from '@ember/debug';

let deprecations;

registerDeprecationHandler((message, options, next) => {
  if (deprecations) {
    deprecations.push(message);
  } else {
    next(message, options);
  }
});

export default function(hooks) {
  hooks.beforeEach(function() {
    deprecations = [];
    this.deprecations = deprecations;
  });

  hooks.afterEach(function() {
    deprecations = null;
  });
}
```

**Review by sukima — APPROVED** [review]

**Pull request merged**
